git-srep/src/main.rs

```diff
@@ -0,0 +1,184 @@
+use clap::Parser;
+use colored::Colorize;
+use git_utils_shared as git_utils;
+use rayon::prelude::*;
+use std::process::Command;
+
+/// Replace given phrase inside files of the provided paths.
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    /// The word to substitute
+    search: String,
+    /// The word to substitute with
+    replace: String,
+
+    #[clap(
+        long,
+        short,
+        takes_value = false,
+        help = "Substituted word in git dirty files"
+    )]
+    force: bool,
+
+    #[clap(required = true)]
+    paths: Vec<String>,
+}
+
+fn is_file_dirty(file_path: &str) -> bool {
+    let modified = git_utils::file::file_is_modified(file_path);
+    let untracked = git_utils::file::file_is_untracked(file_path);
+    if modified {
+        println!("modified -> {}", file_path.red());
+    }
+    if untracked {
+        println!("Untracked -> {}", file_path.red())
+    }
+
+    modified || untracked
+}
+
+fn change_word_in_file(file_path: &str, search: &str, replace: &str) {
+    println!(
+        "Replacing {} -> {} in {}",
+        search,
+        replace,
+        file_path.green()
+    );
+    let change_success = Command::new("sed")
+        .args(["-i", &format!("s/{search}/{replace}/g"), file_path])
+        .status()
+        .expect("Failed ")
+        .success();
+
+    if !change_success {
+        panic!(
+            "Error in replacing {} -> {} in {}",
+            search,
+            replace,
+            file_path.red()
+        )
+    }
+}
+
+fn main() {
+    let args = Args::parse();
+    println!("{:?}", args);
+    // let files = git_utils::get_files_with_word_in_it(&args.search);
+
+    let files: Vec<String> = args
+        .paths
+        .par_iter()
+        .map(|path| {
+            if args.force {
+                git_utils::get_files_with_word_in_it_not_git(&args.search, path)
+            } else {
+                git_utils::get_files_with_word(&args.search, path)
+            }
+        })
+        .filter(|file| file.is_some())
+        .map(|files| files.unwrap())
+        .flatten()
+        .collect();
+
+    println!("{files:?}");
+    if files.is_empty() {
+        println!("There is not any file containing the given word or all files are not staged. Try running 'git add .' ");
+        std::process::exit(1);
+    }
+
+    let dirty_files = files.par_iter().any(|file| is_file_dirty(file));
+    if dirty_files && !args.force {
+        println!(
+            "There is a dirty file (untracked or modified), can't perform replacement. Use --force to ovveride this feature"
+        )
+    } else {
+        files
+            .par_iter()
+            .for_each(|file| change_word_in_file(file, &args.search, &args.replace))
```

Comment:
`sed` should take all of the arguments at once, e.g. `sed -i 's/search/replace/g' -- file1 file2 file3 file4`.

Reply:
Done in bf8bedf30e5c7f33c23860893f2454720f957143